crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -120,6 +123,99 @@ impl<'a> DataRef<'a> {
 // Type aliases for lock gaurds
 type SharedWriteGuard<T> = ArcRwLockWriteGuard<RawRwLock, T>;
 type SharedMutexGuard<T> = ArcMutexGuard<RawMutex, T>;
+type SharedReadGuard<T> = ArcRwLockReadGuard<RawRwLock, T>;
+
+#[allow(dead_code)]
+pub struct TxId {
+    committed_state_shared_lock: SharedReadGuard<CommittedState>,
+    lock_wait_time: Duration,
+    timer: Instant,
+}
+
+#[allow(dead_code)]
+impl TxId {
+    fn table_id_from_name(&self, table_name: &str, database_address: Address) -> super::Result<Option<TableId>> {
+        self.committed_state_shared_lock
+            .iter_by_col_eq(
+                &ExecutionContext::internal(database_address),
+                ST_TABLES_ID,
+                NonEmpty::new(StTableFields::TableName.col_id()),
+                AlgebraicValue::String(table_name.to_owned()),
+            )
+            .map(|mut iter| {
+                iter.next()
+                    .map(|row| TableId(*row.view().elements[0].as_u32().unwrap()))
+            })
+    }
+
+    fn iter<'a>(&'a self, ctx: &'a ExecutionContext, table_id: &TableId) -> super::Result<Iter<'a>> {
```

Comment:
I'm unclear on the relationship between a `TxId` and an `ExecutionContext`. What happens if I perform multiple operations on a single `TxId` with different `ExecutionContext`s? Is it possible that `TxId` should hold an `&ExecutionContext`? I assume you've thought about this and come up with a good solution, but I'd like to see comments which explain your reasoning added to the code.

Reply:
I'm not entirely sure if I understood correctly, but it seems like there might be a need to refactor the usage of `ExecutionContext` in the datastore. Currently, it doesn't mostly consider the `WorkloadType` (sql/reducer), often defaulting to Internal.
So for example, If you call `Locking::table_id_from_name_mut_tx` it doesn't take `context` as param and uses `Internal` but `iters` do.

Currently, I've made `Tx` implementation consistent with `MutTxId`, but it might be worth to revisit it as whole as a separate task .

> What happens if I perform multiple operations on a single TxId with different ExecutionContexts

This may mess up with labels on metrics for `WorkLoadType` but will still provide meaningfull values as they are local states.

Reply:
Yes, method arguments are consistent between `MutTxId` and `TxId`. So, If some method does not takes an `ExecutionContext`, it's `TxId` counter-part will also not and vice versa.